### ISSUE TITLE
Selecting a WiFi Connection that device is currently connected to crashes Easer

### ISSUE THREAD

**Issue by dantuck** [renyuneyun/Easer#188]
From Conditions create a new condition and select your currently connected WiFi access point using either Match SSID or Match BSSID. After save the application crashes. When you start the application again the the status on Outline is "Not Running".

**Comment by renyuneyun** [comment]
Thanks for reporting. This seems strange and doesn't happen on my side.
Could it be related to permission?

If not, can you tell me your phone and ROM (&version)?

**Comment by dantuck** [comment]
I checked the permissions and also reinstalled as a safe precaution. I have tested a number of times but once saved while connected to the WiFi network the application crashes. I have enabled logging and attached the csv. Hope that helps.

[logs_0.csv.zip](https://github.com/renyuneyun/Easer/files/2499573/logs_0.csv.zip)

Model: Pixel
Android version: 9
Build number PPR2.181005.003

**Comment by dantuck** [comment]
I received and update this morning and the app started to run which is much better than it was before. Selecting a WiFi connection that I am currently connected too no longer causes a crash. I will test the condition today to ensure it is triggered.

**Comment by Eiyeron** [comment]
Hello. I can replicate this on my phone since my upgrade to Android Pie. I was wondering what made the crash happen and indeed I can reproduce it by connecitng my phone to a registered wifi network.

Model : Nokia 6.1 TA-1043 (2018)
Android version : 9 (stock ROM, Android One compliant)
Build number : 00WW_3_260

**Comment by melkhatt** [comment]
Same here
Mi mix 2 - Android 9
Easer 0.7.1.1 & 0.7.2 alpha

**Comment by Eiyeron** [comment]
On my side, yes, it seems so.

**Comment by enboig** [comment]
I also have the same problem, but I think it not related to saving network, it happens when detects the network connection. My steps are:
1. Disable wifi
2. Add SSID manually and save without problems
3. Enable Wifi
4. Easer crahes
I am also using Pie 9 on Android One

**Comment by renyuneyun** [comment]
Since I saw the crash log is finally working, would you mind copying / uploading the log produced when the crash happens?
This really feels strange (I can't reproduce on my Android 9.0 device) and some more technical information should be helpful.

**Comment by enboig** [comment]
Ok, I created a rule to mute volume when I arrive home. It check wifi.
```
BUILD_CONFIG: {}
APP_VERSION_CODE: 113
USER_CRASH_DATE: 2019-07-14T15:07:55.186+02:00
ANDROID_VERSION: 9
BRAND: xiaomi
PHONE_MODEL: Mi A2 Lite
PRODUCT: daisy
STACK_TRACE: java.lang.RuntimeException: Error receiving broadcast Intent { act=android.net.wifi.STATE_CHANGE flg=0x4000010 (has extras) } in ryey.easer.e.e.j.a@571ecdb
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0(LoadedApk.java:1401)
	at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run(Unknown Source:2)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6694)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.net.wifi.WifiInfo.getSSID()' on a null object reference
	at ryey.easer.e.e.j.b.a(WifiConnSlot.java:7)
	at ryey.easer.e.e.j.b.a(WifiConnSlot.java:4)
	at ryey.easer.e.e.j.b.a(WifiConnSlot.java:2)
	at ryey.easer.e.e.j.a.onReceive(WifiConnSlot.java:7)
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0(LoadedApk.java:1391)
	... 8 more
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.net.wifi.WifiInfo.getSSID()' on a null object reference
	at ryey.easer.e.e.j.b.a(WifiConnSlot.java:7)
	at ryey.easer.e.e.j.b.a(WifiConnSlot.java:4)
	at ryey.easer.e.e.j.b.a(WifiConnSlot.java:2)
	at ryey.easer.e.e.j.a.onReceive(WifiConnSlot.java:7)
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0(LoadedApk.java:1391)
	at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run(Unknown Source:2)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6694)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```
To stop the bootloop of the app, I have to disable wifi, and delete the rule.

**Comment by renyuneyun** [comment]
This ought to be fixed in d3e04dc90ec5f706ffad48515f3dc9cd2b1bca13 , and should appear soon.

Thanks for reporting!
